src/services/read-from-disk.service.js

```diff
@@ -206,14 +211,24 @@ export function loadAllProjectDependencies(
   return loadPackageJson(projectPath).then(
     packageJson =>
       new Promise((resolve, reject) => {
-        const dependencyNames = Object.keys(packageJson.dependencies);
+        // Check for existence of both dependencies and devDependencies
+        // We can reasonably assume all projects have dependencies
+        // but some may not have devDependencies
+        const deps = Object.keys(packageJson.dependencies);
+        const devDeps = packageJson.devDependencies
+          ? Object.keys(packageJson.devDependencies)
+          : [];
+
+        // If devDeps exists, add to dependencyNames, otherwise don't
+        const dependencyNames = devDeps ? [...deps, ...devDeps] : [...deps];
 
         // Each project in a Guppy directory should have a package.json.
         // We'll read all the project info we need from this file.
         asyncMap(
           dependencyNames,
           function(dependencyName, callback) {
-            loadProjectDependency(projectPath, dependencyName)
+            // We pass in devDeps to run a location check
+            loadProjectDependency(projectPath, dependencyName, devDeps)
```

Comment:
Right, yeah this is a bit tricky.

The only issue I have with your solution is that `loadProjectDependency` shouldn't need to know about any other dependencies; it's more information than it needs, and I like when functions are given the minimum necessary to do their job.

So I'd move the check up a level, to happen outside, and just pass in a `location` string:

```js
asyncMap(
  dependencyNames,
  function(dependencyName, callback) {
    // No `devDeps &&` check necessary, since it's guaranteed to be an array
    const dependencyLocation = devDeps.includes(dependencyName)
      ? 'devDependencies'
      : 'dependencies';

    loadProjectDependency(projectPath, dependencyName, dependencyLocation)
  }
)
```

And then `loadProjectDependency` would look like:

```js
export function loadProjectDependency(
  projectPath: string,
  dependencyName: string,
  dependencyLocation: 'dependencies' | 'devDependencies',
) {
  // ...stuff

  const dependency = {
    location: dependencyLocation,
  };
}
```

Reply:
Yeah this is much nicer, same result but easier to read 👌